controllers/users.js

```diff
@@ -0,0 +1,65 @@
+const model = require ('../models/users')
+
+module.exports.getAll = function getAll(req, res){
+  model.find({}).exec((err, data) => {
+    if (err) return res.status(404).send()
+    res.json({ data })
+  })
+}
+
+module.exports.newUser = function newUser(req, res) {
+  let alreadyTaken = false 
+  const newUser = new model(req.body)
+  model.find({userName:newUser.userName}).exec((error,data) => {
+    if (error) return console.error(error)
+    for (var i = data.length - 1; i >= 0; i--) {
+      if(data[i].userName === newUser.userName){
+        alreadyTaken = true
+        return res.json({message: 'already taken'})
+      }
+    }
+
+    if(alreadyTaken === false) {
+      newUser.save((err, data) => {
+        if (err) return res.status(500).send()
+        res.json({ id: data._id })
+      })
+    }
+  })
+}
+
+module.exports.deleteTrue =  function deleteTrue(req, res) {
+  const newUser = new model(req.body)
```

Comment:
same

Reply:
Here is different, I need to know the index of the model that matches the header so I can change the data, what can be usefull is to use the findOneAndUpdate.
I do use that function sucessfully!